cpp/tests/CMakeLists.txt

```diff
@@ -38,28 +47,97 @@ add_library(cugraphtestutil STATIC
             ../../thirdparty/mmio/mmio.c)
 
 target_compile_options(cugraphtestutil
-            PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
-                   "$<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CUDA>:${CUGRAPH_CUDA_FLAGS}>>"
+    PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
+           "$<$<COMPILE_LANGUAGE:CUDA>:${CUGRAPH_CUDA_FLAGS}>"
 )
 
 set_property(TARGET cugraphtestutil PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 target_include_directories(cugraphtestutil
     PUBLIC
-        "${CMAKE_CURRENT_SOURCE_DIR}/../../thirdparty/mmio"
         "${CMAKE_CURRENT_SOURCE_DIR}"
+        "${CMAKE_CURRENT_SOURCE_DIR}/../../thirdparty/mmio"
+    PRIVATE
         "${CUGRAPH_SOURCE_DIR}/src"
 )
 
 target_link_libraries(cugraphtestutil
     PUBLIC
         cugraph::cugraph
-        NCCL::NCCL
     PRIVATE
-        cuco::cuco
         GTest::gtest
 )
 
+
+###################################################################################################
+# - common C API test utils -----------------------------------------------------------------------
+
+add_library(cugraph_c_testutil STATIC
+            c_api/test_utils.cpp)
+
+target_compile_options(cugraph_c_testutil
+    PUBLIC "$<$<COMPILE_LANGUAGE:C>:${CUGRAPH_C_FLAGS}>"
+           "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
+           "$<$<COMPILE_LANGUAGE:CUDA>:${CUGRAPH_CUDA_FLAGS}>"
+)
+
+target_include_directories(cugraph_c_testutil
+    PRIVATE
+        "${CUGRAPH_SOURCE_DIR}/src"
+)
+
+set_property(TARGET cugraph_c_testutil PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_link_libraries(cugraph_c_testutil
+    PUBLIC
+        CUDA::toolkit
+    PRIVATE
+        cugraph::cugraph_c
+        cugraph::cugraph
+)
+
+if(BUILD_CUGRAPH_MG_TESTS)
+    include(../cmake/thirdparty/get_nccl.cmake)
```

Comment:
Another module we could move into the tests directory if NCCL is no longer linked to by cugraph itself.

Reply:
I am ambivalant about `get_nccl.cmake` being in the normal `cmake` folder. We do the same for other test conditional dependencies like GTest